livekit-agents/livekit/agents/voice/agent_activity.py

```diff
@@ -1353,8 +1368,9 @@ def _on_first_frame(_: asyncio.Future) -> None:
                 )
             elif len(new_fnc_outputs) > 0:
                 # add the tool calls and outputs to the chat context even no reply is generated
-                self._agent._chat_ctx.items.extend(new_calls)
-                self._agent._chat_ctx.items.extend(new_fnc_outputs)
+                for msg in tool_messages:
+                    msg.created_at = reply_started_at
```

Comment:
We can just reuse the `find_insertion_index`, this is safer, otherwise we must ensure there is no async calls in-between, and this is too hard to guarantee (Imagine a user adding a new message)